### DISCUSSION TITLE
Does JavaScript Promise.all() run in parallel or sequential? | Leo's dev blog – stories, insights, and ideas

### DISCUSSION THREAD

**Discussion by giscus** [hta218/leohuynh.dev#62] [Announcements]
# Does JavaScript Promise.all() run in parallel or sequential? | Leo's dev blog – stories, insights, and ideas

JavaScript is a single-threaded programming language, so it can't run multiple things at the same time, Promise.all() actually runs promises concurrently, not in parallel!. Let's dive in to see how it works.

https://www.leohuynh.dev/blog/does-promise-all-run-in-parallel-or-sequential

**Comment by hta218** [comment]
#### Hey, thanks for checking out this post! 💡

If you found it useful (or have some ideas for improvement), I’d love to hear from you! Leave a comment below with your thoughts, or give a reaction to show your support. It means a lot to me!

Cheers 🍺
